tracing/span.go

```diff
@@ -16,6 +16,13 @@ const (
 	SpanTypeServer
 )
 
+type contextKey int
+
+const (
+	serverSpanKey contextKey = iota
```

Comment:
Is `contextKey iota` a good idea? I remember starting out like this at a previous project and then I would run into context key conflicts across services pretty soon, after that I switched to strings. This might not apply here.

Reply:
As long as they are different types, this shouldn't be a problem. Say you have:

```go
type foo int
type bar int
f = foo(0)
b = bar(0)
```

`f` & `b` do not equal because they have different types, that's how context keys work.